sklearn/linear_model/_base.py

```diff
@@ -340,7 +343,8 @@ def _rescale_data(X, y, sample_weight):
     sample_weight = np.asarray(sample_weight)
     if sample_weight.ndim == 0:
         sample_weight = np.full(n_samples, sample_weight, dtype=sample_weight.dtype)
-    sample_weight = np.sqrt(sample_weight)
+    if square_root:
+        sample_weight = np.sqrt(sample_weight)
```

Comment:
Instead of another parameter, what do you think of always returning `np.sqrt(sample_weight)` along with the scaled X and y?

Reply:
Can we go with the new parameter now and think about that again after I looked into the other linear models ?
Since the issue seems similar in other models, we might have to refactor this helper anyway.

Reply:
Makes sense. Call the parameter `sqrt_sample_weights` instead of `square_root`?